src/rail/estimation/algos/deepdisc.py

```diff
@@ -143,11 +145,41 @@ class DeepDiscInformer(CatInformer):
     def __init__(self, args, comm=None):
         CatInformer.__init__(self, args, comm=comm)
 
+    def inform(self, training_data, metadata):
+        self.set_data('input', training_data)
+        self.set_data('metadata', metadata)
+        self.run()
+        self.finalize()
+        return self.get_handle('model')
+
+    def finalize(self):
+        self.temp_dir.cleanup()
+
     def run(self):
         """
         Train a inception NN on a fraction of the training data
         """
         train_data = self.get_data("input")
+        metadata = self.get_data("metadata")
+
+        # create an iterator here
+        itr = self.input_iterator("input")
+        self.temp_dir = tempfile.TemporaryDirectory()
+        for start_idx, _, chunk in itr:
+            for idx, image in enumerate(chunk):
+                this_img_metadata = metadata[start_idx + idx]
+                height = this_img_metadata["height"]
+                width = this_img_metadata["width"]
+
+                #! This spot could have bugs width, height, or height, width???
+                reformed_image = image.reshape(6, width, height).astype(np.float32)
+
+                filename = f'image_{start_idx + idx}.npy'
+                file_path = os.path.join(self.temp_dir.name, filename)
+                np.save(file_path, reformed_image)
+
+                # we want the dictionary associated with this particular image.
+                metadata[start_idx + idx]['filename'] = file_path
 
         cfgfile = self.config.cfgfile
         batch_size = self.config.batch_size
```

Comment:
We can use the `Param(<type>, <default_value>, required=<bool>, msg=<help string>)` at the beginning of this class definition to define reasonable default values, and help clarify what the expected input dictionary needs to contain. 
See this as an example: https://github.com/LSSTDESC/rail_pzflow/blob/main/src/rail/estimation/algos/pzflow_nf.py#L75-L100

Reply:
Tracked in https://github.com/lincc-frameworks/deepdisc/issues/75